xwiki-commons-core/xwiki-commons-observation/xwiki-commons-observation-local/src/main/java/org/xwiki/observation/internal/DefaultObservationManager.java

```diff
@@ -221,13 +221,15 @@ public void addListener(EventListener eventListener)
     @Override
     public void removeListener(String listenerName)
     {
-        synchronized (this.listenersByEvent) {
-            getListenersByName().remove(listenerName);
-            for (Map.Entry<Class<? extends Event>, Map<String, RegisteredListener>> entry : this.listenersByEvent
-                .entrySet()) {
-                entry.getValue().remove(listenerName);
-                if (entry.getValue().isEmpty()) {
-                    this.listenersByEvent.remove(entry.getKey());
+        if (this.listenersByEvent != null) {
```

Comment:
The other option is to not have to check for this and initialize the listenersByEvent as some empty map. I don't recall why we do this. At first glance, since there's only a single instance of DefaultObservationManager it doesn't seem like it would be a costly operation to do a `new ConcurrentHashMap<>();`.